matrix/src/permuted.rs

```diff
@@ -0,0 +1,82 @@
+use core::ops::Deref;
+
+use p3_field::PackedValue;
+
+use crate::dense::RowMajorMatrix;
+use crate::Matrix;
+
+/// A RowPermutation remaps row indices, and can change the height.
+pub trait RowPermutation: Send + Sync {
+    fn height(&self) -> usize;
+    fn permute_row_index(&self, r: usize) -> usize;
```

Comment:
Maybe some of the language should be tweaked since this isn't really a permutation (e.g. when striding). Maybe `map_`, `convert_` or `transform_` in place of `permute_`?